src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Legacy/RoslynCSharpTokenizer.cs

```diff
@@ -58,6 +61,26 @@ public RoslynCSharpTokenizer(SeekableTextReader source, CSharpParseOptions parse
     internal override void StartingBlock()
     {
         _roslynTokenParser.SkipForwardTo(Source.Position);
+        ResetIsOnlyWhitespaceOnLine();
+    }
+
+    private void ResetIsOnlyWhitespaceOnLine()
+    {
+        // Reset isOnlyWhitespaceOnLine for the new block
+        _isOnlyWhitespaceOnLine = true;
+        for (var i = Source.Position - 1; i >= 0; i--)
```

Comment:
Consider being explicit that `i` is an `int` - the loop wouldn't work on unsigned integers.